data_pipeline/generic_web_api/generic_web_api_data_etl.py

```diff
@@ -81,20 +81,9 @@ def get_newline_delimited_json_string_as_json_list(json_string):
 # pylint: disable=fixme,too-many-arguments
 def get_data_single_page(
         data_config: WebApiConfig,
-        cursor: Optional[str] = None,
-        from_date: Optional[datetime] = None,
-        until_date: Optional[datetime] = None,
-        page_number: Optional[int] = None,
-        page_offset: Optional[int] = None,
+        url_compose_arg: UrlComposeParam,
```

Comment:
Refactoring this to make the part that generates those parameters easier to test.